### ISSUE TITLE
removePackageFields fails when removing lifecycle scripts (i.e. `prepack`)

### ISSUE THREAD

**Issue by dreamorosi** [lerna-lite/lerna-lite#637]
### Describe the bug

When using the `publish.removePackageFields` utility, and using it to remove the `scripts` field, if a package has `prepack` or `postpack` scripts the command crashes.

### Expectation

Ideally I would expect Lerna lite to read any prepack/postpack commands, cache them, remove the fields, and then still run them.

If that's not possible (I realize it might be complex, especially if you have a script referencing another), at least I would expect these script to be ignored (possibly with a warning). 

### Reproduction

1. Initialize a new package
2. Add a `script.prepack` command
3. Run the publish command with `publish.removePackageFields` set to remove the `scripts` field

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/@lerna-lite/cli/schemas/lerna-schema.json",
  "packages": [
    "packages/parameters",
  ],
  "version": "1.9.1",
  "npmClient": "npm",
  "message": "chore(release): %s [skip ci]",
  "command": {
    "version": {
      "conventionalCommits": true,
      "forcePublish": false,
      "commitHooks": false,
      "syncWorkspaceLock": true
    },
    "publish": {
      "removePackageFields": [
        "publishConfig",
        "scripts",
        "lint-staged",
        "devDependencies"
      ]
    }
  }
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
23 verbose mutation Removed "scripts" field from "@aws-lambda-powertools/parameters" package.
....
69 error TypeError: Cannot read properties of undefined (reading 'prepack')
69 error     at runLifecycle (file:///home/ec2-user/prep/node_modules/@lerna-lite/core/dist/utils/run-lifecycle.js:56:50)
69 error     at packDirectory (file:///home/ec2-user/prep/node_modules/@lerna-lite/publish/dist/lib/pack-directory.js:26:11)
69 error     at async file:///home/ec2-user/prep/node_modules/p-pipe/index.js:10:19
```
</p></details>

### Environment Info

System:
    OS: Linux 6.1 Amazon Linux 2023
    CPU: (2) arm64 Neoverse-N1
    Memory: 2.94 GB / 3.73 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.15.0 - /run/user/1000/fnm_multishells/280239_1686582997446/bin/node
    npm: 9.5.0 - /run/user/1000/fnm_multishells/280239_1686582997446/bin/npm
  npmPackages:
    @lerna-lite/publish: ^2.4.1 => 2.4.1 
    @lerna-lite/version: ^2.4.1 => 2.4.1 

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by dreamorosi** [comment]
I have tried forking & then cloning the repo and following the contributing guidelines but was not able to get the unit tests run successfully. Also the guideline mentions an outdated command `format:write` which I think is now `prettier:write`.

<img width="650" alt="image" src="https://github.com/lerna-lite/lerna-lite/assets/7353869/7cd66da9-6b1a-49a9-8014-0ad54d74d767">

I'm not sure I can contribute without a working baseline, appreciate the opportunity tho.

Feel free to close the issue if you'd like.

**Comment by ghiscoding** [comment]
Running the unit tests one at a time is an issue with Vitest and I raised an issue on their project [here](https://github.com/vitest-dev/vitest/issues/3129) but it doesn't look like they'll fix that anytime soon. If you run all the tests, not just one, then it should pass, also the only thing that makes them fail is quote escaping (sometime it uses single quote, sometime double, sometime none). I know it's a bit of a pain but since I switched the project to ESM but I got live with it and like I said it always passes if you test all the tests at once... You can contribute and I can take care of the tests if that's the only problem, usually what I do is simply to ignore the quote escaping when testing one test file at a time, then when I'm all done with my tests I ran a full test all and make sure it passes this way, not ideal but it works.

> Also the guideline mentions an outdated command `format:write` which I think is now `prettier:write`.

ah yeah I renamed the npm script, you are correct

**Comment by ghiscoding** [comment]
fixed it in PR #647 by keeping any lifecycle scripts in place, in other words even if you declare `--remove-package-fields 'scripts'` but it found a lifecycle scripts (ie `prepack`, ...)  it will keep that one in place but remove all other ones.

I also found a way to fix all unit tests that were failing when ran one at time, this will hopefully make it easier for anyone to contribute in the future. 

The fix was fixed under the new version [2.4.2](https://github.com/lerna-lite/lerna-lite/releases/tag/v2.4.2)

Thanks for the feedback
